cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/repository/InMemoryFhirRepository.java

```diff
@@ -125,6 +125,14 @@ public <B extends IBaseBundle, T extends IBaseResource> B search(
                 .values();
         if (searchParameters == null || searchParameters.isEmpty()) {
             resourceList.forEach(builder::addCollectionEntry);
+        } else if (searchParameters.containsKey("id")) {
+            for (IBaseResource resource : resourceList) {
```

Comment:
Issue: This implementation is incorrect. The point is that the resourceMap is _already_ indexed by Id. You don't need to scan all the resources. You can simply look at the one you need from the index.

Reply:
Issue: The SearchParameter is `_id`, not `id`. The implementation and the tests should reflect that.